packages/headless/src/controllers/insight/breadcrumb-manager/headless-insight-breadcrumb-manager.ts

```diff
@@ -0,0 +1,271 @@
+import {toggleSelectDateFacetValue} from '../../../features/facets/range-facets/date-facet-set/date-facet-actions';
+import {dateFacetSelectedValuesSelector} from '../../../features/facets/range-facets/date-facet-set/date-facet-selectors';
+import {DateFacetValue} from '../../../features/facets/range-facets/date-facet-set/interfaces/response';
+import {NumericFacetValue} from '../../../features/facets/range-facets/numeric-facet-set/interfaces/response';
+import {toggleSelectNumericFacetValue} from '../../../features/facets/range-facets/numeric-facet-set/numeric-facet-actions';
+import {numericFacetSelectedValuesSelector} from '../../../features/facets/range-facets/numeric-facet-set/numeric-facet-selectors';
+import {
+  logInsightStaticFilterDeselect,
+  toggleSelectStaticFilterValue,
+} from '../../../features/static-filter-set/static-filter-set-actions';
+import {
+  StaticFilterSlice,
+  StaticFilterValue,
+} from '../../../features/static-filter-set/static-filter-set-state';
+import {
+  CategoryFacetSection,
+  ConfigurationSection,
+  DateFacetSection,
+  FacetSection,
+  NumericFacetSection,
+  SearchSection,
+} from '../../../state/state-sections';
+import {
+  Breadcrumb,
+  BreadcrumbManager,
+  BreadcrumbManagerState,
+  BreadcrumbValue,
+  buildCoreBreadcrumbManager,
+  CategoryFacetBreadcrumb,
+  DateFacetBreadcrumb,
+  DeselectableValue,
+  FacetBreadcrumb,
+  NumericFacetBreadcrumb,
+  StaticFilterBreadcrumb,
+} from '../../core/breadcrumb-manager/headless-core-breadcrumb-manager';
+import {logFacetBreadcrumb} from '../../../features/facets/facet-set/facet-set-insight-analytics-actions';
+import {executeSearch} from '../../../features/insight-search/insight-search-actions';
+import {logNumericFacetBreadcrumb} from '../../../features/facets/range-facets/numeric-facet-set/numeric-facet-insight-analytics-actions';
+import {logDateFacetBreadcrumb} from '../../../features/facets/range-facets/date-facet-set/date-facet-insight-analytics-actions';
+import {logClearBreadcrumbs} from '../../../features/facets/generic/facet-generic-insight-analytics-actions';
+import {loadReducerError} from '../../../utils/errors';
+
+export type {
+  NumericFacetBreadcrumb,
+  FacetBreadcrumb,
+  DateFacetBreadcrumb,
+  CategoryFacetBreadcrumb,
+  StaticFilterBreadcrumb,
+  Breadcrumb,
+  BreadcrumbValue,
+  BreadcrumbManagerState,
+  BreadcrumbManager,
+  DeselectableValue,
+};
+/**
+ * Creates a `BreadcrumbManager` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @returns A `BreadcrumbManager` controller instance.
+ */
+export function buildBreadcrumbManager(
+  engine: InsightEngine
+): BreadcrumbManager {
+  if (!loadBreadcrumbManagerReducers(engine)) {
+    throw loadReducerError;
+  }
+  const controller = buildCoreBreadcrumbManager(engine);
+  const {dispatch} = engine;
+  const getState = () => engine.state;
+
+  const getBreadcrumbs = <T extends BaseFacetValue>(
```

Comment:
Yeah I think that's a good idea.`getBreadcrumbs` seems generic enough that it can be defined in the core and imported. The other `getXBreadcrumbs` functions are dispatching multiple use-case-specific actions that make it more clunky to have a utils IMO though.